Sources/HTTP/HTTPServer.swift

```diff
@@ -26,15 +26,33 @@ public protocol HTTPServing : class {
 /// abstraction, but the intention is to remove this dependency and reimplement
 /// the class using transport APIs provided by the Server APIs working group.
 public class HTTPServer: HTTPServing {
+
+    /// A subclass for HTTP Options
+    /// HTTPServer to be created on the given `port`, using `handler` to process incoming requests
+    public class Options {
+        /// Note: if port 0 is requested, a random port will be assigned and HTTPServer.port value will
+        /// diverge from HTTPServer.Options.port
+        public let port: Int
+        public let handler: HTTPRequestHandler
```

Comment:
I would keep the handler out of the options. The options should describe the environment. The handler do the handling.